### DISCUSSION TITLE
Cannot run as Windows service but runs interactively

### DISCUSSION THREAD

**Discussion by kamranayub** [ravendb/ravendb#12877] [Q&A] [answered]
I just was upgrading my local 4.2 instance of RavenDB to 5.2.3 and it was going well. I have it all set up as before and I can run Raven.Server.exe interactively using PowerShell:

```
> .\run.ps1

       _____                       _____  ____
      |  __ \                     |  __ \|  _ \
      | |__) |__ ___   _____ _ __ | |  | | |_) |
      |  _  // _` \ \ / / _ \ '_ \| |  | |  _ <
      | | \ \ (_| |\ V /  __/ | | | |__| | |_) |
      |_|  \_\__,_| \_/ \___|_| |_|_____/|____/


      Safe by default, optimized for efficiency

 Build 52012, Version 5.2, SemVer 5.2.3, Commit 999bd2d
 PID 99068, 64 bits, 8 Cores, Phys Mem 31.958 GBytes, Arch: X64
 Source Code (git repo): https://github.com/ravendb/ravendb
 Built with love by Hibernating Rhinos and awesome contributors!
+---------------------------------------------------------------+
Using GC in server concurrent mode.
Node A in cluster 12d52ff8-95dd-496f-bd8f-488be1d68698
Server available on: https://a.xxx.ravendb.community:9092
Tcp listening on 127.0.0.1:9093
Server started, listening to requests...
TIP: type 'help' to list the available commands.
```

Cool!

But when I try to set this up as a service... it bombs and I cannot use service manager to start it 😞 

```sh
> .\setup-as-service.ps1
Server was run before - attempt to use existing configuration.
Service "RavenDB (Next generation NoSQL Database)" has been registered.
E:\ravendb-50\setup-as-service.ps1 : Microsoft.PowerShell.Commands.ServiceCommandException: Failed to start service 'RavenDB (RavenDB)'.
At line:1 char:1
+ .\setup-as-service.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~
```

Any ideas I could try? This is what I see in Service Manager:

![image](https://user-images.githubusercontent.com/563819/138034703-89e88a4f-ad23-4008-9aa1-2cb397f2ceb4.png)

**Comment by kamranayub** [comment]
Workaround for now: run interactively 😄

**Comment by ayende** [comment, marked as answer]
Things to check:
* Is the service path correct? If not, it may be using a different version.
* Are you running as a different user? The service user may not have access to the file system.
* Obvious, don't run interactively and as a service, they will fight for the same ports.
* Look at the event viewer, that should give you any errors.

**Reply by kamranayub** [reply]
🤦‍♂️ It was the user. Needed to run as Local System Account!

![image](https://user-images.githubusercontent.com/563819/138036775-fe50c5b2-2b0d-4bfc-be69-c3b8a2e8c71d.png)
